File: source/manual/ds_writing/writing_java/StepperMotorClass3.java
public void device_factory(String[] devlist) throws DevFailed
{
	for (int i = 0;i < devlist.length;i++)
	{
		Util.out4.println("Device name : " + devlist[i]);
						
		device_list.addElement(new StepperMotor(this,
					  	        devlist[i],
							"A Tango motor",
							DevState.ON,
							"The motor is ON"));

		if (Util.instance()._UseDb == true)
			export_device(((DeviceImpl)(device_list.lastElement())));
		else
			export_device(((DeviceImpl)(device_list.lastElement())),
					devlist[i]);
	}
}

File: source/development/device-api/device-server-writing/writing/steppermotorclass3.cpp
void StepperMotorClass::device_factory(const Tango::_DevVarStringArray *devlist_ptr)
{
	
	for (long i = 0;i < devlist_ptr->length();i++)
	{
		cout4 << "Device name : " << (*devlist_ptr)[i] << endl;
						
		device_list.push_back(new StepperMotor(this,
						       (*devlist_ptr)[i]));							 

		if (Tango::Util::_UseDb == true)
			export_device(device_list.back());
		else
			export_device(device_list.back(),(*devlist_ptr[i]));
	}
}
